src/sagemaker_server.cc

```diff
@@ -440,6 +440,37 @@ SagemakerAPIServer::SageMakerMMEListModel(evhtp_request_t* req)
   evhtp_send_reply(req, EVHTP_RES_OK);
 }
 
+void
+SagemakerAPIServer::SageMakerMMEHandleLoadError(
+    evhtp_request_t* req, TRITONSERVER_Error* load_err)
+{
+  const char* message = TRITONSERVER_ErrorMessage(load_err);
+  std::string error_string(message);
+
+  const std::vector<std::string> error_messages{
+      "CUDA out of memory", /* pytorch */
+      "CUDA_OUT_OF_MEMORY", /* tensorflow */
+      "Out of memory",      /* generic */
+      "out of memory", "MemoryError"};
+
+  EVBufferAddErrorJson(req->buffer_out, load_err);
+
+  for (long unsigned int i = 0; i < error_messages.size(); i++) {
+    if (error_string.find(error_messages[i]) != std::string::npos) {
+      /* Return a 507*/
+      evhtp_send_reply(req, 507);
+      LOG_VERBOSE(1)
+          << "Received an OOM error during LOAD MODEL. Returning a 507.";
+      return;
+    } else {
+      /* Return a 400*/
+      evhtp_send_reply(req, EVHTP_RES_BADREQ);
```

Comment:
Actually this should be done outside the for loop, so it checks if any messages are matched before returning 400 early

Reply:
@nskool let me know when it is ready for a CI run

Reply:
thanks, fixed - request you to run the CI